sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/dynamodb/DynamoDBIO.java

```diff
@@ -342,6 +343,8 @@ public boolean test(Throwable throwable) {
 
     abstract @Nullable SerializableFunction<T, KV<String, WriteRequest>> getWriteItemMapperFn();
 
+    abstract @Nullable List<String> getOverwriteByPKeys();
```

Comment:
Can you please remove the `@Nullable` here and set an empty collection as the default value on the `build()` method. This will simplify the validations in the rest of the code and will make it simpler.